### PR TITLE
[WIP] cinnamon-window-tracker.c: Add support for desktop files from flatpaks

### PR DESCRIPTION
**Pull request by Cobinja**
This adds support for associating windows from flatpaks with their respective .desktop-file.
This is needed for pinning them to window lists like our builtin Grouped Window List.

Depends on https://github.com/linuxmint/muffin/pull/527

### CODE REVIEW

**Comment by Cobinja** [comment]
Force pushed a version with proper flatpak support, which takes into account that flatpak .desktop-files can be renamed from their original upstream ones.

File: src/cinnamon-app-system.c [hunk 1]
```diff
@@ -897,6 +1013,26 @@ cinnamon_app_system_lookup_startup_wmclass (CinnamonAppSystem *system,
...
 
+/**
+ * cinnamon_app_system_lookup_startup_wmclass_for_flatpak_app:
+ * @system: a #CinnamonAppSystem
+ * @wmclass: (nullable): A WM_CLASS value
+ *
+ * Find a valid flatpak application whose .desktop file contains a
+ * StartupWMClass entry matching @wmclass.
+ *
+ * Returns: (transfer none): A #CinnamonApp for @wmclass
```
Comment (itzexor) [inline src/cinnamon-app-system.c]: This should be annotated (nullable) as g_hash_table_lookup can return NULL, as well as the explicit return NULL....unless these are considered error conditions

File: src/cinnamon-app-system.c [hunk 2]
```diff
@@ -897,6 +1013,26 @@ cinnamon_app_system_lookup_startup_wmclass (CinnamonAppSystem *system,
   return g_hash_table_lookup (system->priv->startup_wm_class_to_app, wmclass);
 }
 
+/**
+ * cinnamon_app_system_lookup_startup_wmclass_for_flatpak_app:
+ * @system: a #CinnamonAppSystem
+ * @wmclass: (nullable): A WM_CLASS value
```
Comment (itzexor) [inline src/cinnamon-app-system.c]: this probably shouldn't be labeled nullable (or optional) if we just return NULL when it is NULL

File: src/cinnamon-app-system.c [hunk 3]
```diff
@@ -658,18 +677,63 @@ on_apps_tree_changed_cb (GMenuTree *tree,
...
+        GList *flatpak_renamed_from, *l;
+        GHashTableIter flatpakIter;
+        gpointer flatpakKey, flatpakValue;
+
+        flatpak_renamed_from = cinnamon_app_get_flatpak_renamed_from (app);
+        l = flatpak_renamed_from;
+        while (l)
+          {
+            g_hash_table_replace (self->priv->desktop_filename_to_flatpak_app,
+                                  (gchar*)(l->data), app);
```
Comment (itzexor) [inline src/cinnamon-app-system.c]: possible lost ref on app?

Comment (Cobinja) [reply src/cinnamon-app-system.c]: Resolved with https://github.com/linuxmint/cinnamon/pull/8701/commits/e58b0f94d29643876782beeec9210809cae2b53d

**Comment by Cobinja** [comment]
Closed in favor of https://github.com/linuxmint/cinnamon/pull/9705

**Pull request closed without merging**